gneiss/regression/_ols.py

```diff
@@ -185,17 +177,74 @@ class OLSModel(RegressionModel):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+
     def fit(self, **kwargs):
         """ Fit the model.
 
         Parameters
         ----------
         **kwargs : dict
            Keyword arguments used to tune the parameter estimation.
-
         """
-        # assumes that the underlying submodels have implemented `fit`.
-        self.results = [s.fit(**kwargs) for s in self.submodels]
+        Y = self.response_matrix
+        X = self.design_matrices
+        n, p = X.shape
+        inv = np.linalg.pinv(np.dot(X.T, X))
+        cross = np.dot(inv, X.T)
+        beta = np.dot(cross, Y)
+        pX = np.dot(X, beta)
+        resid = (Y - pX)
+        sst = (Y - Y.mean(axis=0))
+        sse = (resid**2).sum(axis=0)
```

Comment:
Are you treating each of the response variables separately in the least squares estimation? If yes, then your matrix equation is good. Or are you aiming at the multivariate response least squares estimation?